src/python/geoclaw/etopotools.py

```diff
@@ -123,3 +124,95 @@ def etopo1_download(xlimits, ylimits, dx=0.0166666666667, dy=None, \
         topo.read(file_path, topo_type=3)
         return topo
 
+def etopo1_download_nc(xlimits, ylimits, coarsen=1, verbose=False):
+
+    """
+    Download a subset of the etopo1 topography from the NCEI NetCDF server,
+    return it as a topotools.Topography object. 
+
+    The data for points in the region specified by `xlimits, ylimits`
+    is selected and then coarsened by a factor `coarsen`.
+
+    Uses the database described at
+        https://www.ngdc.noaa.gov/mgg/global/global.html
+
+    The lat-lon values are grid-registered as described at
+        https://www.ngdc.noaa.gov/mgg/global/gridregistration.html
+
+    Sample usage:
+
+        from clawpack.geoclaw import etopotools
+        xlimits = [-126,-122]; ylimits = [46,49]
+        topo = etopotools.etopo1_download_nc(xlimits,ylimits,
+                                             coarsen=2,verbose=True)
+        # to plot:
+        topo.plot()
+
+        # to save topofile for input to GeoClaw:
+        topo.write('etopo_sample_2min.tt3', topo_type=3, Z_format='%.0f')  
+
+    This should give a 2-minute resolution DEM of the Western Washington coast.
+    Note that etopo1 Z values are integers (vertical resolution is 1 meter)
+    and using `Z_format='%.0f'` will save as integers to minimize file size.
+    """
+
+    from clawpack.geoclaw import topotools
+    from pylab import find
+    try:
+        import xarray
+    except:
+        print('*** You need to install xarray, see:')
+        print('***     https://xarray.pydata.org/en/stable/')
+
+    x1, x2 = xlimits
+    y1, y2 = ylimits
+
+    assert (type(coarsen) is int) and (coarsen > 0), \
+           '*** coarsen must be a positive integer'
+
+    etopo1_url = 'https://www.ngdc.noaa.gov/thredds/dodsC/global/ETOPO1_Ice_g_gmt4.nc'
```

Comment:
Maybe provide a way to over ride this?  I mostly thinking of temporary future proofing.